src/libmongoc/tests/unified/result.c

```diff
@@ -565,6 +567,50 @@ result_check (result_t *result,
             goto done;
          }
       }
+
+      if (error_response) {
+         if (!result->reply) {
+            test_set_error (
+               error, "%s", "expected error with a reply, but reply unset");
+            goto done;
+         }
+
+         /* Write operations have the raw response wrapped in a errorReplies
+          * array. Read operations return the raw response as the reply. */
+         bson_iter_t iter;
+         bson_t doc_to_match;
+
+         if (bson_iter_init_find (&iter, result->reply, "errorReplies")) {
+            bson_lookup_doc_null_ok (
+               result->reply, "errorReplies.0", &doc_to_match);
```

Comment:
Continuing the conversation in https://github.com/mongodb/mongo-c-driver/pull/1144#discussion_r1021030892

Sorry, but I misread `bson_lookup_doc_null_ok` and thought it gracefully allowed for missing values. I don't think there's any reason to use that, as it could still abort if the array is empty.

I could have sworn there was a function that allowed full field paths and gracefully handled missing keys. Perhaps I was thinking of `bson_iter_find_descendant`, which `bson_lookup` uses internally before it aborts on a failed resolution.

Perhaps it'd be best to use `assert_match_bson()` to check that "errorReplies.0" is a document. That function will emit a test error on failure, unlike `ASSERT_MATCH()`, which aborts. I believe you can use the `$$type`, although I don't know offhand if the function supports field paths (e.g. "errorReplies.0"), or if you might need to do something like:

```
{ "errorReplies": { "0": { "$$type": "object" } } }
```

/cc @kevinAlbs

Reply:
I just pushed changes that check if `errorReplies.0` is a document. I ended up using `BSON_ITER_HOLDS_DOCUMENT` and `BSON_ITER_HOLDS_ARRAY` to check that `errorReplies` was the expected type. I thought to use these because that is how the reply is also checked right above my code (line 496 where `error_labels_contain` is checked).